File: app/src/test/java/com/annushkaproject/programmerscalculator/ProgrammerTests/ProgrammerOrTest.java
package com.annushkaproject.programmerscalculator.ProgrammerTests;

import com.annushkaproject.programmerscalculator.model.ProgrammerCalcModel;
import com.annushkaproject.programmerscalculator.model.WordLength;
import com.annushkaproject.programmerscalculator.utils.ProgrammerOperationsUtil;

import org.junit.Test;

import java.math.BigDecimal;

import static com.annushkaproject.programmerscalculator.model.Operator.OR;
import static org.junit.Assert.*;

public class ProgrammerOrTest {
    @Test
    public void orTestAll_isCorrect() {
        for (WordLength mode : WordLength.values()) {
            assertEquals(0, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(0), new BigDecimal(0), OR, mode)));
            assertEquals(1, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(1), new BigDecimal(0), OR, mode)));
            assertEquals(-1, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(-1), new BigDecimal(1), OR, mode)));
            assertEquals(-1, ProgrammerOperationsUtil.calculateWithData(new ProgrammerCalcModel(new BigDecimal(-1), new BigDecimal(-1), OR, mode)));
        }
    }
}

File: app/src/test/java/com/annushkaproject/programmerscalculator/ProgrammerTests/ProgrammerOrTest.kt
package com.annushkaproject.programmerscalculator.ProgrammerTests

import com.annushkaproject.programmerscalculator.model.Operator
import com.annushkaproject.programmerscalculator.model.ProgrammerCalcModel
import com.annushkaproject.programmerscalculator.model.int_size_enum
import com.annushkaproject.programmerscalculator.utils.ProgrammerOperationsUtil.calculateWithData
import org.junit.Assert
import org.junit.Test
import java.math.BigDecimal

class ProgrammerOrTest {
    @Test
    fun orTestAll_isCorrect() {
        for (mode in int_size_enum.values()) {
            Assert.assertEquals(0, calculateWithData(ProgrammerCalcModel(BigDecimal(0), BigDecimal(0), Operator.OR, mode)))
            Assert.assertEquals(1, calculateWithData(ProgrammerCalcModel(BigDecimal(1), BigDecimal(0), Operator.OR, mode)))
            Assert.assertEquals(-1, calculateWithData(ProgrammerCalcModel(BigDecimal(-1), BigDecimal(1), Operator.OR, mode)))
            Assert.assertEquals(-1, calculateWithData(ProgrammerCalcModel(BigDecimal(-1), BigDecimal(-1), Operator.OR, mode)))
        }
    }
}